File: livingrimoire start here/LivinGrimoire java/older livingrimoire var/old java_core/DiSkillV2.java
package com.yotamarker.lgkotlin1;

public class DiSkillV2 extends AbsCmdReq {
    protected Kokoro kokoro; // consciousness, shallow ref class to enable interskill communications
    protected DISkillUtils diSkillUtils = new DISkillUtils();
    protected Algorithm outAlg = null; // skills output

    public DiSkillV2(Kokoro kokoro) {
        super();
        this.kokoro = kokoro;
    }

    @Override
    public void input(String ear, String skin, String eye) {
    }
    @Override
    public final void output(Neuron noiron) {
        if (outAlg != null) {
            noiron.algParts.add(outAlg);
            outAlg = null;
        }
    }
}

File: livingrimoire start here/LivingGrimoire Kotlin/old LG kotlin version/kotlin_core_txt/LGCore/DiSkillV2.kt
package com.yotamarker.lgkotlinfull.LGCore

class DiSkillV2(  // consciousness, shallow ref class to enable interskill communications
        protected var kokoro: Kokoro) : AbsCmdReq() {
    protected var diSkillUtils = DISkillUtils()
    protected var outAlg: Algorithm? = null // skills output
    override fun input(ear: String, skin: String, eye: String) {}
    override fun output(noiron: Neuron) {
        if (outAlg != null) {
            noiron.algParts.add(outAlg!!)
            outAlg = null
        }
    }
}